chainer/functions/connection/multihead_attention.py

```diff
@@ -0,0 +1,286 @@
+import typing as tp  # NOQA
+
+import chainer
+from chainer.functions.activation import softmax
+from chainer.functions.array import concat
+from chainer.functions.array import expand_dims
+from chainer.functions.array import split_axis
+from chainer.functions.array import repeat
+from chainer.functions.array import reshape
+from chainer.functions.array import transpose
+from chainer.functions.array import where
+from chainer.functions.math import average
+from chainer.functions.math import matmul
+from chainer.functions.noise import dropout as dropout_func
+from chainer.functions.connection import linear
+from chainer import types  # NOQA
+from chainer import variable
+
+
+InputType = tp.Union[variable.Variable, types.NdArray]
+
+
+def generate_zeros(device, shape, dtype):
+    with chainer.using_device(device):
+        return device.xp.zeros(shape=shape, dtype=dtype)
+
+
+def generate_ones(device, shape, dtype):
+    with chainer.using_device(device):
+        return device.xp.ones(shape=shape, dtype=dtype)
+
+
+def multihead_attention(
+    n_head,  # type: int
+    embedding_size,  # type: int
+    query,  # type: InputType
+    key,  # type: InputType
+    value,  # type: InputType
+    proj_in_W,  # type: tp.Union[variable.Variable, tp.Tuple[variable.Variable, variable.Variable, variable.Variable]]  # NOQA
+    proj_in_b=None,  # type: variable.Variable
+    bias_k=None,  # type: variable.Variable
+    bias_v=None,  # type: variable.Variable
+    proj_out_W=None,  # type: variable.Variable
+    proj_out_b=None,  # type: variable.Variable
+    add_zero_attn=False,  # type: bool
+    dropout=0,  # type: float
```

Comment:
```suggestion
    attention_dropout=0,  # type: float
```
Use the name used in `L.MultiHeadAttention` and fix `from chainer import dropout as dropout_func` to `from chainer.functions import dropout`